gui/ros_gui/gui/configpanel/configpanel.py

```diff
@@ -0,0 +1,357 @@
+import threading
+from pathlib import Path
+import os
+import time
+import json
+
+from rcl_interfaces.msg import ParameterType, Parameter
+from rcl_interfaces.srv import SetParametersAtomically, GetParameters, ListParameters
+from rclpy.node import Node
+
+from nicegui import Client, ui, app
+from .local_file_picker import local_file_picker
+
+CONF_PATH = os.path.join(os.path.expanduser('~'), ".astazero/ATOS/conf")
+MAX_TIMEOUT = 1
+
+class ConfigPanelNode(Node):
+    """ This node is responsible for rendering the config panel and setting parameters in ATOS nodes.
+    """
+
+    def __init__(self) -> None:
+        """ Initializes the node, fetches all ros parameters and renders the config panel with these values.
+        """
+        super().__init__('config_panel')
+        schema_path = os.path.join(CONF_PATH, "atos-param-schema.json")
+        self.json_schema = json.load(open(schema_path, "r"))
+        modules = self.json_schema["modules"].keys()
+        time.sleep(0.5) # Allow time for this node to initialize before discovering other nodes, otherwise it will fail to find the others.
+        self.active_nodes_and_namespaces = self.get_node_names_and_namespaces()
+        self.active_node_list = [node for node, namespace in self.active_nodes_and_namespaces if node in modules and "atos" in namespace]
+        self.active_node_list = list(set(self.active_node_list)) # Removes duplicates
+
+        self.client_list = self.init_clients(self.active_node_list)
+        self.parameters = {}
+        threading.Thread(target=self.get_parameters_list, args=(self.client_list,)).start()
+        with Client.auto_index_client:
+            pass
+
+        @ui.page(path='/config', title="ATOS Config Panel")
+        def render_configpanel() -> None:
+            with ui.splitter(value=30).classes('w-1/2') as self.splitter:
+                with self.splitter.before:
+                    with ui.tabs().props('vertical').classes('w-fit') as tabs:
+                        ui.tab('Home', icon='🏠')
+                        for node in self.active_node_list:
+                            ui.tab(node.replace("_", " "))
+                with self.splitter.after:
+                    with ui.tab_panels(tabs, value='Home'):
+                        with ui.tab_panel('Home'):
+                            ui.label('Welcome to ATOS config panel!').classes('text-h4')
+                            if self.active_node_list:
+                                ui.label("Select a node to configure. Press the refresh button if you can't find the node you're looking for.")
+                            else:
+                                ui.label("No nodes were discovered. Press the refresh button to try again.")
```

Comment:
Hmm, strange. I haven't had this issue but I added an automatic refresh if no nodes were found. Let me know if it helps, because I can't test the behavior myself